MVVMIOS/Models/Response.swift

```diff
@@ -7,6 +7,44 @@
 
 import Foundation
 
-struct Response: Codable {
+struct Response: Codable, Equatable {
     let name: String
 }
+
+struct News<T: Codable>: Codable {
```

Comment:
Shouldn't need any generics on the Codable structs. Just `struct Name: Codable {` should be enough.

Reply:
@tinder-owenthomas 
Done removed the generic Codable, But what if the response has same pattern let's say, the server sends everything wrapped inside the data property! What would be the best approach. For example,

```swift

struct Response: Codable {
    let status: Int
     let data: <Some Codable>
    let message: String
}

struct AnotherResponse: Codable {
        let status: Int
     let data: <Some Codable>
    let message: String
}

Would not passing the data as generic Codable be a good approach to avoid duplication. What do you think here?

Reply:
Each type in a Codable should be a known type. There should be no need (that I'm aware of) for a generic "Response" object. Every endpoint that you hit on an API should have its own custom Codable struct to process the response, and if that response just has a `data` element at the top level which represents the actual response then you'd have something like
```
struct NewsResponse: Codable {
    let data: [NewsItem]
}

struct NewsItem: Codable {
    let id: String?
    let title: String?
    // etc.
}
```